build_tools/benchmarks/reporting/parse_shark_benchmarks.py

```diff
@@ -0,0 +1,308 @@
+            # benchmark that is not included in the snapshot yet, emit a
+            # warning.
+            print(
+                "Warning: No baseline results found for " + model + ", " +
+                dialect + ", " + data_type + ", " + device +
+                ". Using IREE version as baseline. Please update baseline csv.")
+            engine = iree_results.engine.iloc[0]
+            baseline_latency = iree_results.loc[iree_results.engine == engine]
+            baseline_latency = baseline_latency.iloc[0]["ms/iter"]
+          else:
+            engine = baseline_results.engine.iloc[0]
+            baseline_latency = baseline_results.loc[baseline_results.engine ==
+                                                    engine]
+            baseline_latency = baseline_latency.iloc[0]["ms/iter"]
+
+          iree_latency = iree_results.loc[iree_results.engine == "shark_iree_c"]
+          iree_latency = iree_latency.iloc[0]["ms/iter"]
+          iree_vs_baseline = html_utils.format_latency_comparison(
+              iree_latency, baseline_latency)
+
+          if df_shark is not None:
+            shark_results = df_shark.loc[(df_shark.model == model) &
+                                         (df_shark.dialect == dialect) &
+                                         (df_shark.data_type == data_type) &
+                                         (df_shark.device == device)]
+
+            shark_latency = shark_results.loc[shark_results.engine ==
+                                              "shark_iree_c"]
+            shark_latency = shark_latency.iloc[0]["ms/iter"]
+            shark_vs_baseline = html_utils.format_latency_comparison(
+                shark_latency, baseline_latency)
+            iree_vs_shark = html_utils.format_latency_comparison(
+                iree_latency, shark_latency)
+          else:
+            # If there are no SHARK benchmarks available, use default values.
+            # These columns will be hidden later.
+            shark_latency = 0
+            shark_vs_baseline = "1.0x faster"
+            iree_vs_shark = "1.0x faster"
+
+          summary.loc[len(summary)] = [
+              model, engine, data_type, dialect, device,
+              "{:.1f}".format(baseline_latency), "{:.1f}".format(iree_latency),
+              "{:.1f}".format(shark_latency), iree_vs_baseline,
+              shark_vs_baseline, iree_vs_shark
+          ]
+
+  summary = summary.round(2)
+
+  st = summary.style.set_table_styles(html_utils.get_table_css())
+  st = st.hide(axis="index")
+  if df_shark is None:
+    st = st.hide_columns(
+        subset=[_SHARK_LATENCY, _SHARK_VS_BASELINE, _IREE_VS_SHARK])
+  st = st.set_caption(title)
+  st = st.applymap(html_utils.style_performance, subset=_PERF_COLUMNS)
+  st = st.set_properties(subset=[_MODEL],
+                         **{
+                             "width": "300px",
+                             "text-align": "left",
+                         })
```

Comment:
Why are you creating a dictionary only to immediately unpack it? I think this would be equivalent, right?

```suggestion
  st = st.set_properties(subset=[_MODEL], width="300px", text-align="left")
```

Reply:
It doesn't work because the hyphen in "text-align" causes Python to read it as subtracting two variables.